### ISSUE TITLE
Attributes on functions supplied by decorators trigger `no-member`

### ISSUE THREAD

**Issue by socketpair** [pylint-dev/pylint#9246]
### Bug description

```python
from functools import cache


class SomeClass:
    # Yes, the order of decorators is correct.
    @classmethod
    @cache
    def __cached_fun(cls, arg: int) -> str:  # no-member
        return str(arg)

    @classmethod
    def cache_clear(cls):
        cls.__cached_fun.cache_clear()
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
a.py:13:8: E1101: Method '__cached_fun' has no 'cache_clear' member (no-member)
```

### Expected behavior

No message above. Actually everything works as expected at runtime.

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.4
Python 3.11.6 (main, Oct  3 2023, 00:00:00) [GCC 13.2.1 20230728 (Red Hat 13.2.1-1)]
```

### OS / Environment

Fedora 38
